File: bunnies-and-lozenges/Assets/CS_Scripts/WinText.cs
// handles showing the "you win!" message

using UnityEngine;
using System.Collections;

public class WinText : MonoBehaviour {

	public static bool textOn;
	public static string message;
	
	void Start () {
		textOn = false;
		guiText.text = "";
	}
	
	void Update () {
		if (textOn) {
			guiText.enabled = true;
			guiText.text = message;
			Wait();
		}
	}
	
	// function for waiting
	// waits 2 seconds
	IEnumerator Wait() {
		yield return new WaitForSeconds(2);
	}
}

File: Assets/Scripts/WinText.js
static var textOn:boolean;
static var message:String;
private var timer:float = 0.0;

function Start() {
	timer = 0.0;
	textOn = false;
	guiText.text = "";
}

function Update () {
	if (textOn) {
		guiText.enabled = true;
		guiText.text = message;
		timer += Time.deltaTime;
	}
	
	if (timer >= 5) {
		textOn = false;
		guiText.enabled = false;
		timer = 0.0;
	}
}